src/pages/api/comments.ts

```diff
@@ -0,0 +1,154 @@
+import { getSession } from "@auth0/nextjs-auth0";
+import { Prisma } from "@prisma/client";
+import { withSentry } from "@sentry/nextjs";
+import { NextApiHandler } from "next";
+
+import { prisma } from "../../shared/prisma-helper";
+import { generateAvatarUrl } from "../../shared/user-helpers";
+import {
+  CommentsApiHandlerSuccessfulGetResponseBody,
+  CommentsApiHandlerSuccessfulPostResponseBody,
+  NewComment,
+  PublicCommentInfo,
+} from "../../types";
+
+const commentSelect = {
+  accidentId: true,
+  author: {
+    select: {
+      name: true,
+      email: true,
+      picture: true,
+    },
+  },
+  createdAt: true,
+  id: true,
+  isPublished: true,
+  text: true,
+} as const;
+
+const convertSelectedCommentToPublicInfo = ({
+  accidentId,
+  author,
+  createdAt,
+  id,
+  isPublished,
+  text,
+}: Prisma.CommentGetPayload<{
+  select: typeof commentSelect;
+}>): PublicCommentInfo => ({
+  accidentId,
+  authorAvatarUrl: generateAvatarUrl(author.picture ?? undefined, author.email),
+  authorName: author.name,
+  createdAt: createdAt.toISOString(),
+  id,
+  isPublished,
+  text,
+});
+
+const handler: NextApiHandler = async (req, res) => {
+  const session = getSession(req, res);
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment -- external user type is `any`
+  const myId = session?.user.sub;
+
+  if (req.method === "POST") {
+    if (typeof myId !== "string") {
+      res.status(401).json({
+        error: "not_authenticated",
+        description:
+          "The user does not have an active session or is not authenticated",
+      });
+
+      return;
+    }
+
+    let newComment: NewComment | undefined;
+
+    try {
+      const { accidentId, text } = JSON.parse(req.body as string) as Record<
+        keyof NewComment,
+        unknown
+      >; // type casting is tolerable as we are inside try/catch
+
+      if (
```

Comment:
This is third if deep, I would use some helpers for more readability and simple testing.

Reply:
Yeah you are right. I can split the API route into two to make the logic simpler. We can extract helpers, but if they are only used in one place, they won’t necessarily make things easier.

Reply:
- ➡️ https://github.com/dtpstat/website/pull/72